### PR TITLE
Bump rust toolchain to 2018-03-24.

### PR DESCRIPTION
**Pull request by mrowqa**
Newer toolchain contains [bugfix](https://github.com/rust-lang/rust/pull/49242) required by #14902.

---

- [X] `./mach build -d` does not report any errors
It reports some warnings about deprecated items in [rust-webvr-api](https://github.com/servo/rust-webvr/).
- [X] `./mach build-geckolib` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20589)

### CODE REVIEW

**Comment by Mrowqa** [comment]
Taking a look at [this](https://ci.appveyor.com/project/servo/servo/build/1.0.26249#L771) it seems like `NonZero` i `Zeroable` have been removed, so I have left only the manual implementation of it.

File: components/msg/constellation_msg.rs [hunk 1]
```diff
@@ -353,15 +353,3 @@ impl PartialEq<BrowsingContextId> for TopLevelBrowsingContextId {
...
-pub const TEST_PIPELINE_INDEX: PipelineIndex = unsafe { PipelineIndex(NonZero::new_unchecked(5678)) };
-#[cfg(feature = "unstable")]
-pub const TEST_PIPELINE_ID: PipelineId = PipelineId { namespace_id: TEST_NAMESPACE, index: TEST_PIPELINE_INDEX };
-#[allow(unsafe_code)]
-#[cfg(feature = "unstable")]
-pub const TEST_BROWSING_CONTEXT_INDEX: BrowsingContextIndex =
-    unsafe { BrowsingContextIndex(NonZero::new_unchecked(8765)) };
-#[cfg(feature = "unstable")]
-pub const TEST_BROWSING_CONTEXT_ID: BrowsingContextId =
-    BrowsingContextId { namespace_id: TEST_NAMESPACE, index: TEST_BROWSING_CONTEXT_INDEX };
```
Comment (Mrowqa) [inline components/msg/constellation_msg.rs]: Failed with:
```
error[E0015]: calls in constants are limited to constant functions, struct and enum constructors
   --> components/msg/constellation_msg.rs:358:71
    |
358 | pub const TEST_PIPELINE_INDEX: PipelineIndex = unsafe { PipelineIndex(NonZero::new_unchecked(5678)) };
    |                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0015]: calls in constants are limited to constant functions, struct and enum constructors
   --> components/msg/constellation_msg.rs:364:35
    |
364 |     unsafe { BrowsingContextIndex(NonZero::new_unchecked(8765)) };
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors
```

**Comment by SimonSapin** [comment]
This will likely hit the same WebBluetooth test failures as https://github.com/servo/servo/pull/20474#issuecomment-377291900 which still need to be investigated.

**Comment by SimonSapin** [comment]
By the way #20474 already takes care of the nonzero stuff.

**Comment by Mrowqa** [comment]
Oh, I haven't noticed this PR. In this case my change looks redundant.

**Pull request closed without merging**